src/planscape/impacts/views.py

```
@@ -143,6 +144,18 @@ def perform_create(self, serializer):
             **serializer.validated_data,
         )
 
+    def destroy(self, request, *args, **kwargs):
```

wait do we actually wanna do this? can't we cancel the job instead?

---

Canceling the jobs would require bound the list of tasks triggered to the Treatment Plan, which is more complex.
The simplest solution is preventing the deletion while it is running.

---

I understand it's the easiest solution but we're exposing a very internal error to the user, and blocking the user to complete the action, which is not great.
Maybe we can catch the errors on celery if we don't wanna keep a list of triggered jobs?
I have no idea how much effort this is, so if its out of scope or lower priority, maybe track this somewhere?

---

Well, we can suppress the error on the celery task.
We need to be careful on that, as the success callback triggers an email.
What do you think @george-silva ?

---

not at all, in fact my concern is actually showing this error to the user. We shouldn't be communicating this to the user, is an implementation detail on our side.
I understand if this is something we do not wanna do right now, but ultimately I don't think this is a good way of solving this (blocking the user action). We could instead either cancel the jobs, or catch the errors that happen downstream when the resource is no longer present